### PR TITLE
Feature/transition laptop request 1

### PR DESCRIPTION
**Pull request by Abdalrazaq94**
## Naming Rules

##Transition Laptop Request Status (1)

## Description

creating post + PUT + Delete / laptop_assignment 
post endpoint    api/laptop_assignment
Put :   api/put/laptop_assignment/assignmentId
delete: api/delete/laptop_assignment/assignmentId

## Related to
you may need to run npm recreate-db to start it 

Fixes #1

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: server/api.js [hunk 1]
```diff
@@ -90,5 +90,49 @@ router.get("/laptop_donation", async (req, res) => {
...
+			res.status(400).json({ success: " was not   success" });
+		});
+});
+
+
+
+router.put("/laptop_assignment/:assignmentId", function (request, response) {
+	const assignmentId = request.params.assignmentId;
+	if (assignmentId !== undefined) {
+		db.query("UPDATE laptop_assignment SET status=$3", [
```
Comment (gregdyke) [inline server/api.js]: the update needs a where clause

File: server/api.js [hunk 2]
```diff
@@ -90,5 +90,49 @@ router.get("/laptop_donation", async (req, res) => {
...
+	} else {
+		response.status(404).send("status not found");
+	}
+});
+
+
+router.delete("/laptop_assignment/:assignmentId", function (request, response) {
+	const assignmentId = request.params.assignmentId;
+
+	db.query("DELETE FROM laptop_assignment WHERE id=$3", [assignmentId])
```
Comment (gregdyke) [inline server/api.js]: assignmentId is in position 0 -> it will replace placeholder $1, not $3

File: server/api.js [hunk 3]
```diff
@@ -90,5 +90,49 @@ router.get("/laptop_donation", async (req, res) => {
...
+router.post("/laptop_assignment", (req, res) => {
+
+	let laptopRequestsId = req.body.laptop_request_id;
+	let laptopDonationId = req.body.laptop_donation_id;
+	let status = req.body.status;
+	let assignmentId = req.body.laptop_assignment;
+	const query =
+		" insert into laptop_assignment (laptop_request_id, laptop_donation_id, status) values ($1, $2, $3 )";
+
+	db.query(query, [status, laptopRequestsId, laptopDonationId])
```
Comment (gregdyke) [inline server/api.js]: the array here is in the wrong order

File: server/api.js (hunk 2 above)
Comment (Abdalrazaq94) [reply server/api.js]: I fixed to be $1

File: server/api.js (hunk 3 above)
Comment (Abdalrazaq94) [reply server/api.js]: I have changed it to the right order

File: server/api.js (hunk 1 above)
Comment (Abdalrazaq94) [reply server/api.js]: I have updated it

File: server/api.js [hunk 4]
```diff
@@ -91,4 +91,40 @@ router.get("/laptop_donation", async (req, res) => {
 	}
 });
 
+// post laptop assignment
+router.post("/laptop_assignment", (req, res) => {
+	let laptopRequestsId = req.body.laptop_request_id;
```
Comment (gregdyke) [inline server/api.js]: laptopRequestId

File: server/api.js [hunk 5]
```diff
@@ -91,4 +91,40 @@ router.get("/laptop_donation", async (req, res) => {
...
+	let laptopDonationId = req.body.laptop_donation_id;
+	let status = req.body.status;
+	let assignmentId = req.params.laptop_assignment;
+	const query =
+		" insert into laptop_assignment (laptop_request_id, laptop_donation_id, status) values ($1, $2, $3 )";
+
+	db.query(query, [laptopRequestsId, laptopDonationId, status])
+		.then(() => res.send(`assignmentId ${assignmentId} posted!`))
```
Comment (gregdyke) [inline server/api.js]: you need to get the id back from the rows of the query. Something like `.then((queryResult) => res.send(res.rows[0]))`

File: server/api.js [hunk 6]
```diff
@@ -91,4 +91,40 @@ router.get("/laptop_donation", async (req, res) => {
 	}
 });
 
+// post laptop assignment
+router.post("/laptop_assignment", (req, res) => {
+	let laptopRequestsId = req.body.laptop_request_id;
+	let laptopDonationId = req.body.laptop_donation_id;
+	let status = req.body.status;
+	let assignmentId = req.params.laptop_assignment;
```
Comment (gregdyke) [inline server/api.js]: there is no laptop_assignment in the params -> this is part of the point of a POST request: we use it when we don't have an id yet

File: server/api.js [hunk 7]
```diff
@@ -91,4 +91,40 @@ router.get("/laptop_donation", async (req, res) => {
...
+			console.error(error);
+			res.status(400).json({ success: " was not success" });
+		});
+});
+
+router.put("/laptop_assignment/:assignmentId", function (req, res) {
+	const assignmentId = res.params.assignmentId;
+	const newStatus = req.body.status;
+
+	db.query("UPDATE laptop_assignment SET id =$1", [newStatus, assignmentId])
```
Comment (gregdyke) [inline server/api.js]: still need a where clause. Something like `SET status = $1 WHERE id = $2`

**Review by gregdyke — CHANGES_REQUESTED** [review]

File: server/api.js (hunk 4 above)
Comment (Abdalrazaq94) [reply server/api.js]: fixed

File: server/api.js (hunk 6 above)
Comment (Abdalrazaq94) [reply server/api.js]: Done

File: server/api.js (hunk 5 above)
Comment (Abdalrazaq94) [reply server/api.js]: done

**Review by gregdyke — APPROVED** [review]

**Pull request merged**